### PR TITLE
feat: [PL-38922]: Enabled necessary DW metrics in CG Manager

### PR DESCRIPTION
**Pull request by Raghav-Murali**
## Describe your changes

In this PR we will be enabling the necessary DW metrics in CG manager and disabling the unnecessary ones.

I have tested the changes locally.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48206)

### CODE REVIEW

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48206)* status: 1 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`360-cg-manager/src/main/resources/metrics/metricDefinitions/cg_manager_service_metrics.yaml` line 117 at r1](https://reviewable.io/reviews/harness/harness-core/48206#-NW63yKvACBRUt7KLv5q:-NW63yKvACBRUt7KLv5r:b-1v24q2) ([raw file](https://github.com/harness/harness-core/blob/50533d56466f424f49c62e6217f62304fff67b6d/360-cg-manager/src/main/resources/metrics/metricDefinitions/cg_manager_service_metrics.yaml#L117)):*
> ```YAML
>     type: LastValue
>     unit: count
>   - metricName: currentgen_manager_io_dropwizard_jetty_MutableServletContextHandler_active_dispatches
> ```

@Raghav-Murali Do you think these 3 metrics are helpful? I am not clear on them based on the description.

currentgen_manager_io_dropwizard_jetty_MutableServletContextHandler_active_dispatches
currentgen_manager_io_dropwizard_jetty_MutableServletContextHandler_active_suspended
currentgen_manager_io_dropwizard_jetty_MutableServletContextHandler_active_requests

**Review by Raghav-Murali — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48206)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`360-cg-manager/src/main/resources/metrics/metricDefinitions/cg_manager_service_metrics.yaml` line 117 at r1](https://reviewable.io/reviews/harness/harness-core/48206#-NW63yKvACBRUt7KLv5q:-NW64l0t0RatOJ0ghper:bgiui16) ([raw file](https://github.com/harness/harness-core/blob/50533d56466f424f49c62e6217f62304fff67b6d/360-cg-manager/src/main/resources/metrics/metricDefinitions/cg_manager_service_metrics.yaml#L117)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

@Raghav-Murali Do you think these 3 metrics are helpful? I am not clear on them based on the description.

currentgen_manager_io_dropwizard_jetty_MutableServletContextHandler_active_dispatches
currentgen_manager_io_dropwizard_jetty_MutableServletContextHandler_active_suspended
currentgen_manager_io_dropwizard_jetty_MutableServletContextHandler_active_requests


</blockquote></details>

Basically it captures the number of incoming requests that are being handled or suspended. I think maybe `currentgen_manager_io_dropwizard_jetty_MutableServletContextHandler_active_requests` this is not that useful.

I thought that number of requests currently being handled and suspended might come in handy to determine whether there are enough threads or pods available to handle the requests in a timely manner. Let me know otherwise, thanks.

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48206)* status: 1 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`360-cg-manager/src/main/resources/metrics/metricDefinitions/cg_manager_service_metrics.yaml` line 117 at r1](https://reviewable.io/reviews/harness/harness-core/48206#-NW63yKvACBRUt7KLv5q:-NW69nEJ5BOuHSFILHw5:b-r3m8jf) ([raw file](https://github.com/harness/harness-core/blob/50533d56466f424f49c62e6217f62304fff67b6d/360-cg-manager/src/main/resources/metrics/metricDefinitions/cg_manager_service_metrics.yaml#L117)):*
<details><summary><i>Previously, Raghav-Murali wrote…</i></summary><blockquote>

Basically it captures the number of incoming requests that are being handled or suspended. I think maybe `currentgen_manager_io_dropwizard_jetty_MutableServletContextHandler_active_requests` this is not that useful.

I thought that number of requests currently being handled and suspended might come in handy to determine whether there are enough threads or pods available to handle the requests in a timely manner. Let me know otherwise, thanks.
</blockquote></details>

I think Queue size and Thread Pool Utilization tells us if requests are piling up and there aren't enough threads to process requests

**Review by Raghav-Murali — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48206)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`360-cg-manager/src/main/resources/metrics/metricDefinitions/cg_manager_service_metrics.yaml` line 117 at r1](https://reviewable.io/reviews/harness/harness-core/48206#-NW63yKvACBRUt7KLv5q:-NW6EnPv1DoS5UXwnMiz:b-yj8gej) ([raw file](https://github.com/harness/harness-core/blob/50533d56466f424f49c62e6217f62304fff67b6d/360-cg-manager/src/main/resources/metrics/metricDefinitions/cg_manager_service_metrics.yaml#L117)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

I think Queue size and Thread Pool Utilization tells us if requests are piling up and there aren't enough threads to process requests
</blockquote></details>

Yeah, that makes sense. I will remove these 3.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48206#-:-NW6HLbS4ylEGyBjqVzS:bnfp4nl)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Comment by Raghav-Murali** [comment]
trigger uts

**Comment by Raghav-Murali** [comment]
trigger messagecheck

**Comment by Raghav-Murali** [comment]
trigger ut5

**Comment by Raghav-Murali** [comment]
trigger ut5

**Comment by Raghav-Murali** [comment]
trigger uts

**Comment by Raghav-Murali** [comment]
trigger ut1
